### PR TITLE
HPCC-28430 Not set port number in redirect URL when not available

### PR DESCRIPTION
**Pull request by wangkx**
When ESP receives a request with an invalid session ID, ESP will redirect the request to the Session Login page using the URL from the HTTP Host header in the request. If the URL does not contain a port number, ESP should not add a port number.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by afishbeck — APPROVED** [review]

File: esp/bindings/http/platform/httptransport.ipp [hunk 1]
```diff
@@ -151,6 +152,7 @@ public:
     const char *queryHost(){return m_host.get();}
     virtual void setPort(int port) {m_port = port;};
     virtual int getPort() {return m_port;};
+    virtual bool getHasPortInHost() const {return hasPortInHost;};
```
Comment (ghalliday) [inline esp/bindings/http/platform/httptransport.ipp:155]: minor: extra ; after the function definition (same as setPort and other functions above).
Also, this doesn't need to be virtual - it isn't in a base class or overridden.

**Review by ghalliday — COMMENTED** [review]
@wangkx happy to merge.  Do you want to merge as is, or address minor comments?

**Comment by wangkx** [comment]
I want to address minor comments before merge. After I fix them, I will let you know.

File: esp/bindings/http/platform/httptransport.ipp (hunk 1 above)
Comment (wangkx) [reply esp/bindings/http/platform/httptransport.ipp:155]: Too many old places with the extra ; and virtual. I will only clean them from line 153, 154, and 155.

**Review by ghalliday — APPROVED** [review]

**Comment by wangkx** [comment]
@ghalliday I cleaned the 3 lines of the code and tested. Should be ready to be merged.

**Pull request merged**